feat(curriculum): add smart campus workshop to JS cert (#69120)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/blocks/workshop-smart-campus-directory/6a6a77f5e36f8ea726089905.md

---
id: 6a6a77f5e36f8ea726089905
title: Step 1
challengeType: 1
dashedName: step-1

> # demoType can either be 'onClick' or 'onLoad'. If the project or lab doesn't have a preview, delete the property

Reviewer: ```suggestion
```

> demoType: onClick

Reviewer: ```suggestion
demoType: onLoad
```

---

# --description--

> 

Reviewer: ```suggestion
```

You are building a Smart Campus Directory. The campus data comes to you as a single, deeply nested JavaScript 

> object called `rawData`, which has already been provided for you in the seed code.

Reviewer: ```suggestion
In this workshop, you will work with a smart campus directory and create lookup tables from nested data. You'll use these lookup tables to perform fast and efficient searches. The `rawData` has been provided for you.
```

Take a moment to look at its shape: each key is a department name, and its value holds a `Dept_code` along with
a `programs` array. Each program then holds its own array of `instructors`.

> Log `rawData` to the console so you can inspect this nested structure before working with it.

Reviewer: ```suggestion
To begin, log `rawData` to the console so you can inspect this nested structure before working with it.
```

# --hints--

Test 1

```js

```

# --seed--

## --seed-contents--

```js
const rawData = {
  "Computer Science": {
    Dept_code: "CSE",
    programs: [
      {
        id: "CSE101",
        name: "Database Management System",
        room: "402",
        instructors: [
          { id: "INS001", name: "Alejandro", email: "alejandro@edu.com" },
          { id: "INS002", name: "Kenji", officeHours: null },
        ],
      },
      {
        id: "CSE102",
        name: "Data Structures",
        room: "205",
        instructors: [
          { id: "INS003", name: "Fatima", email: "fatima@edu.com" },
        ],
      },
    ],
  },
  "Business Administration": {
    Dept_code: "BBA",
    programs: [
      {
        id: "BBA101",
        name: "Financial Accounting",
        room: "201",
        instructors: [{ id: "INS004", name: "Sofia", email: "sofia@edu.com" }],
      },
    ],
  },
};

> console.log(rawData);

Reviewer: this shouldn't be here since you are asking campers to do this part.
